helpers/setURLQueryParam.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+const URL = require('url').URL;
+const utils = require('handlebars-utils');
+const common = require('./lib/common.js');
+const SafeString = require('handlebars').SafeString;
+
+const factory = () => {
+    return function(string, key, value) {
+        if (string instanceof SafeString) {
+            string = string.toString();
+        }
+        if (!utils.isString(string) || !common.isValidURL(string)
+            || !utils.isString(key) || !utils.isString(value)
+            || !(key.length > 0) || !(value.length > 0)) {
+            // Return empty string if this does not appear to be a URL or if the
+            // key/value are not strings of at least 1 character in length
+            return ''
```

Comment:
But if they are empty then we should just return the original string. The behavior seems odd to return empty string.